common/persistence/nosql/nosqlplugin/cassandra/workflow_utils.go

```diff
@@ -57,95 +59,79 @@ func executeCreateWorkflowBatchTransaction(
 		return nil
 	}
 
+	requestRangeID := shardCondition.RangeID
 	requestConditionalRunID := ""
 	if currentWorkflowRequest.Condition != nil {
 		requestConditionalRunID = currentWorkflowRequest.Condition.GetCurrentRunID()
 	}
 	// There can be two reasons why the query does not get applied. Either the RangeID has changed, or
 	// the workflow is already started. Check the row info returned by Cassandra to figure out which one it is.
+	rangeIDMismatch := false
+	actualRangeID := int64(0)
+	currentExecutionAlreadyExists := false
+	var actualExecution map[string]interface{}
+	runIDMismatch := false
+	actualCurrRunID := ""
+	lastWriteVersionMismatch := false
+	actualLastWriteVersion := int64(common.EmptyVersion)
+	stateMismatch := false
+	actualState := int(0)
+	concreteExecutionAlreadyExists := false
+	workflowRequestAlreadyExists := false
+	workflowRequestID := ""
+	requestRowType := int(0)
+	var allPrevious []map[string]interface{}
+
 	for {
 		rowType, ok := previous["type"].(int)
 		if !ok {
 			// This should never happen, as all our rows have the type field.
 			break
 		}
 		runID := previous["run_id"].(gocql.UUID).String()
-
 		if rowType == rowTypeShard {
-			if rangeID, ok := previous["range_id"].(int64); ok && rangeID != shardCondition.RangeID {
-				// CreateWorkflowExecution failed because rangeID was modified
-				return &nosqlplugin.WorkflowOperationConditionFailure{
-					ShardRangeIDNotMatch: common.Int64Ptr(rangeID),
-				}
+			if actualRangeID, ok = previous["range_id"].(int64); ok && actualRangeID != requestRangeID {
+				// UpdateWorkflowExecution failed because rangeID was modified
+				rangeIDMismatch = true
```

Comment:
it's much easier to trace the code with early termination. let's just return WorkflowOperationConditionFailure here as before

Reply:
we need to check the errors in a specific order. This early return doesn't allow us to check the errors in the order we want.

Reply:
ok makes sense.